src/odemis/gui/cont/multi_point_correlation.py

```diff
@@ -0,0 +1,815 @@
+        if event.GetKeyCode() == wx.WXK_RETURN:
+            # Commit the value in the currently edited cell
+            if self.grid.IsCellEditControlEnabled():
+                self.grid.DisableCellEditControl()
+                # Suppress the Enter key from moving to the next row
+            return
+        else:
+            # For other keys, allow the default behavior
+            event.Skip()
+
+    def _update_feature_correlation_target(self, surface_fiducial=False):  # , fm_poi = False):
+        """Populate the correlation target based on the latest changes to save it."""
+        if not self.correlation_target:
+            return
+
+        if surface_fiducial:
+            fib_surface_fiducial = self._tab_data_model.fib_surface_point.value
+            self.correlation_target.fib_surface_fiducial = fib_surface_fiducial
+        else:
+            # Corner case: When fiducials are deleted and the indices are not continiuos. Then the fiducial pairs
+            # will be incorrectly matched together.
+            # TODO Handle the corner case
+            fib_fiducials = []
+            fm_fiducials = []
+            self.correlation_target.fm_pois = []
+            for target in self._tab_data_model.main.targets.value:
+                if "FIB" in target.name.value:
+                    fib_fiducials.append(target)
+                elif "FM" in target.name.value:
+                    fm_fiducials.append(target)
+                elif "POI" in target.name.value:
+                    self.correlation_target.fm_pois = target
+            if fib_fiducials:
+                fib_fiducials.sort(key=lambda x: x.index.value)
+                self.correlation_target.fib_fiducials = fib_fiducials
+            if fm_fiducials:
+                fm_fiducials.sort(key=lambda x: x.index.value)
+                self.correlation_target.fm_fiducials = fm_fiducials
+
+        acq_conf = conf.get_acqui_conf()
+        save_features(acq_conf.pj_last_path, self._tab_data_model.main.features.value)
+
+    def check_correlation_conditions(self) -> bool:
+        """
+        Minimum 4 FIB and FM fiducials and 1 POI in FM are required to run the correlation.
+        :return: (bool) True if the conditions are met, False otherwise
+        """
+        if self.correlation_target:
+            if (len(self.correlation_target.fib_fiducials) >= 4 and len(
+                    self.correlation_target.fm_fiducials) >= 4 and self.correlation_target.fm_pois and len(
+                self._tab_data_model.views.value[0].stream_tree) > 0) and self.correlation_target.fib_stream:
+                return True
+            else:
+                self.correlation_target.reset_attributes()
+                self._tab_data_model.projected_points = []
+                self.correlation_txt.SetLabel("Correlation RMS Deviation :")
+                for vp in self._viewports:
+                    if vp.view.name.value == "SEM Overview":
```

Comment:
Relying on the view name is quite easily broken. It'd better to explicitly use the viewport. Both viewports are assigned as attributes, so you should be able to directly do it. I guess it's `self._panel.vp_correlation_tr`?